app.rb

```diff
@@ -0,0 +1,170 @@
+    option = ''
+    loop do
+      puts "Do you want to create a student (1) or a teacher (2)? [Input the number]\n"
+      option = gets.chomp.to_i
+      break if [1, 2].include?(option)
+    end
+    puts 'Name:'
+    name = gets.chomp
+    puts 'Age:'
+    age = gets.chomp.to_i
+    case option
+    when 1
+      puts 'Has parent permission? [Y/N]'
+      permission = gets.chomp.downcase
+      permission = true if permission == 'y'
+      permission = false if permission == 'n'
+      puts 'Classroom:'
+      classroom = gets.chomp
+      student = Student.new(age, classroom, name, parent_permission: permission)
+      puts "Student #{student.name} has been created successfully."
+    when 2
+      puts 'Specialization:'
+      specialization = gets.chomp
+      teacher = Teacher.new(age, specialization, name)
+      puts "Teacher #{teacher.name} has been created successfully."
+    end
+    puts '*' * 70
+  end
+
+  def create_book
+    puts '*' * 70
+    puts 'Title:'
+    title = gets.chomp
+    puts 'Author:'
+    author = gets.chomp
+    book = Book.new(title, author)
+    puts "Book #{book.title} has been created successfully."
+    puts '*' * 70
+  end
+
+  def create_rental
+    puts '*' * 70
+    if @books.length.zero?
+      (puts 'The book list is empty please create a book first.')
+    elsif @people.length.zero?
+      (puts 'The people list is empty please add a person first.')
+    else
+      puts 'Select a book from the following list by number:'
+      @books.each_with_index do |b, index|
+        puts "#{index + 1} | #{b.title} - By #{b.author}"
+      end
+      book = @books[gets.chomp.to_i - 1]
+      puts 'Select a person from the following list by number:'
+      @people.each_with_index do |p, index|
+        puts "#{index + 1} | Name: #{p.name} - Age: #{p.age}" if p.is_a?(Student)
+        puts "#{index + 1} | Name: #{p.name} - Age: #{p.age}" if p.is_a?(Teacher)
+      end
+      person = @people[gets.chomp.to_i - 1]
+      puts 'Date:'
+      date = gets.chomp
+      Rental.new(date, book, person)
+      puts('Rental created successfully.')
+    end
+    puts '*' * 70
+  end
+
+  def list_rentals
+    puts '*' * 70
+    if @rentals.length.zero?
+      (puts 'No rentals found.')
+    else
+      puts 'Select a person from the following list by number:'
+      @people.each_with_index do |person, index|
+        puts "#{index + 1} | Name: #{person.name} - Age: #{person.age}"
+      end
+      person = @people[gets.chomp.to_i - 1]
+      rental_list = @rentals.select { |rental| rental.person.id == person.id }
+      if rental_list.length.zero?
+        (puts 'No rentals found for this person.')
+      else
+        puts 'Rental list:'
+        rental_list.each_with_index do |rental, index|
+          puts "#{index + 1} | Date: #{rental.date}, Book: #{rental.book.title}- By the author: #{rental.book.author}, Rented by: #{rental.person.name}"
+        end
+      end
+    end
+    puts '*' * 70
+  end
+end
```

Comment:
- [ ] Let's select a person from the list by their `id` instead of number as it is a requirement to list all rentals for a given person by `id`. let's collect the `id` of the person and search the rentals array to filter `rental.person.id` that is equal to the `id` the user added. Then we list all rentals for that given person.